test/services/utils/IssueUtil.test.ts

```diff
@@ -0,0 +1,287 @@
+    });
+    case2.forEach((c) => {
+      expect(updateNotification(c[0], c[1], c[2])).toBe(c[3]);
+    });
+  });
+
+  test("updateStatus function test", () => {
+    const case1: [string, string, string, string][] = [
+      [
+        `
+        tidb
+        tikv
+`,
+        "asuka730",
+        "chanllenge1",
+        `
+        tidb
+        tikv
+
+
+
+## Status
+
+&nbsp;&nbsp;Current challenger: @asuka730
+&nbsp;&nbsp;Current Program: chanllenge1
+<!-- probot:Status -->
+
+`,
+      ],
+    ];
+
+    const case2: [string, string, undefined, string][] = [
+      [
+        `
+        tidb
+        tikv
+`,
+        "asuka730",
+        undefined,
+        `
+        tidb
+        tikv
+
+
+
+## Status
+
+&nbsp;&nbsp;Current challenger: @asuka730
+<!-- probot:Status -->
+
+`,
+      ],
+    ];
+    const case3: [string, undefined, undefined, string][] = [
+      [
+        `
+        tidb
+        tikv
+`,
+        undefined,
+        undefined,
+        `
+        tidb
+        tikv
+
+
+
+## Status
+
+&nbsp;&nbsp; The challenge has not picked yet.
+<!-- probot:Status -->
+
+`,
+      ],
+    ];
+
+    case1.forEach((c) => {
+      expect(updateStatus(c[0], c[1], c[2])).toBe(c[3]);
+    });
+    case2.forEach((c) => {
+      expect(updateStatus(c[0], c[1], c[2])).toBe(c[3]);
+    });
+    case3.forEach((c) => {
+      expect(updateStatus(c[0], c[1], c[2])).toBe(c[3]);
+    });
+  });
+
+  test("ping in an empty issue", async (done) => {
+    const issueBody = {
+      body: notificationHead,
+    };
+
+    // mock Context.issue
+    nock("https://api.github.com")
+      .get("/repos/Rustin-Liu/ti-challenge-bot/issues/1")
+      .reply(200, {
+        owner: "Rustin-Liu",
+        repo: "ti-challenge-bot",
+        number: 1,
+        body: "",
+      });
+
+    nock("https://api.github.com")
+      .patch("/repos/Rustin-Liu/ti-challenge-bot/issues/1", (body: any) => {
+        done(expect(body).toMatchObject(issueBody));
+        return true;
+      })
+      .reply(200);
+    await probot.receive({ name: "issue_comment", payload });
+  });
+
+  test("ping in an normal issue", async (done) => {
+    const issueBody = { body: issueBodyTemplate + notificationHead };
+    // mock Context.issue
+    nock("https://api.github.com")
+      .get("/repos/Rustin-Liu/ti-challenge-bot/issues/1")
+      .reply(200, {
+        owner: "Rustin-Liu",
+        repo: "ti-challenge-bot",
+        number: 1,
+        body: issueBodyTemplate,
+      });
+
+    // mock update operation
+    nock("https://api.github.com")
+      .patch("/repos/Rustin-Liu/ti-challenge-bot/issues/1", (body: any) => {
+        done(expect(body).toMatchObject(issueBody));
+        return true;
+      })
+      .reply(200);
+    await probot.receive({ name: "issue_comment", payload });
+  });
+
+  // I have no idea how to mock the `Context` and DB connection yet.
+
+  // test("pick-up in an issue without mentor information", async (done) => {
```

Comment:
I don't know how to test `pick-up` command.

how to mock the `Context` object, or  just test `give-up/pick-up`?

Reply:
Good job! I will review it tonight or tomorrow.